packages/suite/src/views/wallet/receive/components/FreshAddress.tsx

```diff
@@ -164,15 +161,21 @@ export const FreshAddress = ({
                     <StyledFreshAddress>{addressValue}</StyledFreshAddress>
                 </FreshAddressWrapper>
             </AddressContainer>
-            <StyledButton
-                data-test="@wallet/receive/reveal-address-button"
-                icon="TREZOR_LOGO"
-                onClick={() => showAddress(firstFreshAddress.path, firstFreshAddress.address)}
-                isDisabled={disabled || locked || !firstFreshAddress}
-                isLoading={locked}
+            <Tooltip
+                content={
+                    coinjoinDisallowReveal && <Translation id="RECEIVE_ADDRESS_COINJOIN_DISALLOW" />
+                }
             >
-                <Translation id="RECEIVE_ADDRESS_REVEAL" />
-            </StyledButton>
+                <StyledButton
+                    data-test="@wallet/receive/reveal-address-button"
+                    icon="TREZOR_LOGO"
+                    onClick={() => showAddress(firstFreshAddress.path, firstFreshAddress.address)}
+                    isDisabled={disabled || locked || !firstFreshAddress || coinjoinDisallowReveal}
```

Comment:
also redundant because of https://github.com/trezor/trezor-suite/pull/8301/files#diff-1445aee641015342e651bf45bd0c4c8e2ef6bfe5688ca8daf5585d63592c525dR141
```suggestion
                    isDisabled={disabled || locked || coinjoinDisallowReveal}
```

Reply:
https://github.com/trezor/trezor-suite/pull/8301/commits/c79285caa933f709bb767fb5356e32b1a5422e27